src/gt4py/next/otf/binding/nanobind.py

```diff
@@ -117,16 +123,31 @@ class BindingCodeGenerator(TemplatedGenerator):
     WrapperFunction = as_jinja(
         """\
         decltype(auto) {{name}}(
-            {{"\n,".join(parameters)}}
+            {{"\n,".join(parameters + ["std::optional<nanobind::dict> exec_info"])}}
         )
         {
+            if (exec_info.has_value()) {                
+                exec_info->operator[]("run_cpp_start_time") = static_cast<double>(
+                        std::chrono::duration_cast<std::chrono::nanoseconds>(
+                            std::chrono::high_resolution_clock::now().time_since_epoch()).count())/1e9;
+            }
             {{body}}
+            {% if sync %}
+            cudaDeviceSynchronize();
+            {% endif %}
```

Comment:
not sure why I added this here, probably copied from cartesian, should be just unconditionally in the next if